packages/ember-htmlbars/lib/node-managers/component-node-manager.js

```diff
@@ -65,10 +67,11 @@ ComponentNodeManager.create = function(renderNode, env, options) {
   // Instantiate the component
   component = createComponent(component, isAngleBracket, createOptions, renderNode, env, attrs);
 
-  // If the component specifies its template via the `layout properties
-  // instead of using the template looked up in the container, get them
+  // If the component specifies its layout via the `layout` property
+  // instead of using the template looked up in the container, get it
   // now that we have the component instance.
   layout = get(component, 'layout') || layout;
+  component[HAS_BLOCK] = !!templates.default;
```

Comment:
You need to add it to `createOptions` in [this hash](https://github.com/emberjs/ember.js/blob/12d23bc540369e469bc4813b2dc0579f54cc1a51/packages/ember-htmlbars/lib/node-managers/component-node-manager.js#L48):

```javascript
let createOptions = {
  parentView,
  [HAS_BLOCK]: !!templates.default
};
```